sale_coupon_multi_use/models/sale_coupon_consumption_line.py

```diff
@@ -16,7 +16,8 @@ class SaleCouponConsumptionLine(models.Model):
     sale_order_line_id = fields.Many2one(
         "sale.order.line", "Sale Order Line", required=True, ondelete="cascade"
     )
-    amount = fields.Float()
+    currency_program_id = fields.Many2one(related="coupon_id.program_id.currency_id")
+    amount = fields.Monetary(currency_field="currency_program_id")
```

Comment:
To move in your `sale_coupon_multi_currency` module: it's not linked to multi use.

Reply:
Yes it is linked, it has nothing to do with `sale_coupon_multi_currency`. Also if I would move it there, module would crash, because it does not know anything about sale.coupon.consumption_line model.